datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -1635,6 +1643,87 @@ pub(crate) fn asymmetric_join_output_partitioning(
     }
 }
 
+pub(crate) trait BatchTransformer: Debug + Clone {
```

Comment:
I think some context would help here. Something like this perhaps:

```suggestion
/// Trait for incrementally generating Join output. 
///
/// This trait is used to limit some join outputs 
/// so it does not produce single large batches 
pub(crate) trait BatchTransformer: Debug + Clone {
```